src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToNewComponentCodeActionResolver.cs

```diff
@@ -0,0 +1,146 @@
+    private readonly IClientConnection _clientConnection;
+    public ExtractToNewComponentCodeActionResolver(
+        IDocumentContextFactory documentContextFactory,
+        LanguageServerFeatureOptions languageServerFeatureOptions,
+        IClientConnection clientConnection)
+    {
+        _documentContextFactory = documentContextFactory ?? throw new ArgumentNullException(nameof(documentContextFactory));
+        _languageServerFeatureOptions = languageServerFeatureOptions ?? throw new ArgumentNullException(nameof(languageServerFeatureOptions));
+        _clientConnection = clientConnection ?? throw new ArgumentNullException(nameof(clientConnection));
+    }
+
+    public string Action => LanguageServerConstants.CodeActions.ExtractToNewComponentAction;
+
+    public async Task<WorkspaceEdit?> ResolveAsync(JsonElement data, CancellationToken cancellationToken)
+    {
+        if (data.ValueKind == JsonValueKind.Undefined)
+        {
+            return null;
+        }
+
+        var actionParams = JsonSerializer.Deserialize<ExtractToNewComponentCodeActionParams>(data.GetRawText());
+        if (actionParams is null)
+        {
+            return null;
+        }
+
+        if (!_documentContextFactory.TryCreate(actionParams.Uri, out var documentContext))
+        {
+            return null;
+        }
+
+        var componentDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
+        if (componentDocument.IsUnsupported())
+        {
+            return null;
+        }
+
+        if (!FileKinds.IsComponent(componentDocument.GetFileKind()))
+        {
+            return null;
+        }
+
+        var path = FilePathNormalizer.Normalize(actionParams.Uri.GetAbsoluteOrUNCPath());
+        var directoryName = Path.GetDirectoryName(path) ?? throw new InvalidOperationException($"Unable to determine the directory name for the path: {path}");
+        var templatePath = Path.Combine(directoryName, "Component");
+        var componentPath = FileUtilities.GenerateUniquePath(templatePath, ".razor");
+
+        // VS Code in Windows expects path to start with '/'
+        var updatedComponentPath = _languageServerFeatureOptions.ReturnCodeActionAndRenamePathsWithPrefixedSlash && !componentPath.StartsWith("/")
+            ? '/' + componentPath
+            : componentPath;
+
+        var newComponentUri = new UriBuilder
+        {
+            Scheme = Uri.UriSchemeFile,
+            Path = updatedComponentPath,
+            Host = string.Empty,
+        }.Uri;
+
+        var text = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
+        if (text is null)
+        {
+            return null;
+        }
+
+        var componentName = Path.GetFileNameWithoutExtension(componentPath);
+        var newComponentContent = text.GetSubTextString(new CodeAnalysis.Text.TextSpan(actionParams.ExtractStart, actionParams.ExtractEnd - actionParams.ExtractStart)).Trim();
```

Comment:
Is it possible to add a using directive for `Microsoft.CodeAnalysis.Text` rather than partially qualifying here?